CONTRIBUTING.md

```diff
@@ -5,28 +5,111 @@ This project is very new, so we'll probably revise these guidelines. Please
 comment on a bug before putting significant effort in, if you'd like to
 contribute.
 
-Guidelines:
+## Guidelines
 
 * Follow the Style Guide (see below)
 * Keep work branches in your own github fork, rebase your own branches at will
 * Squash or rebase branches before merging to master so the commits make sense
-  on their own
+on their own
 * Get a SGTM from someone relevant before merging
 * Keep commits to master bisect safe (i.e. each commit should pass all tests)
 * Sign-off commits before merging (see below)
 * Make sure your commit message references the issue or bug number, if there is
-  one, identifies the reviewers, and follows a readable style, with the long
-  description including any additional information that's likely to help future
-  spelunkers.
-  For example:
-  ```
-  Frobnicate the URL bazzer before flattening pilchard. Fixes #6. r=mossop,rnewman
-
-  The frobnication method used is as described in Podder's Miscellany, page 15.
-  Note that this pull request doesn't include tests, because we're bad people.
-
-  Signed-off-by: Random J Developer <random@developer.example.org>
-  ```
+one, identifies the reviewers, and follows a readable style, with the long
+description including any additional information that's likely to help future
+spelunkers (see below)
+
+```
+Frobnicate the URL bazzer before flattening pilchard, r=mossop,rnewman. Fixes #6.
+
+The frobnication method used is as described in Podder's Miscellany, page 15.
+Note that this pull request doesn't include tests, because we're bad people.
+
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
+
+## Example
+
+* Fork this repo.
+
+* Clone your fork locally:
+```
+git clone git@github.com:YOURNAME/tofino.git
+```
+Check your remotes:
+```
+git remote --verbose
+```
+Make sure you have an upstream remote defined:
+```
+git remote add upstream https://github.com/mozilla/tofino
+```
+
+* Create a new branch to start working on a bug or feature:
+```
+git checkout -b some-new-branch
+```
+
+* Do some work, making sure you signoff every commit:
+```
+git commit --signoff --message “Some commit message”
+```
```

Comment:
I think it would be good to add a step:

    * Rebase as you work, so your commits are a logical sequence to read rather than a record of your fevered typing.
    ```
    git rebase -i origin/master
    ```
    Or using `git reset --soft` (as described in [a tale of tree trees](http://www.infoq.com/presentations/A-Tale-of-Three-Trees))